.github/workflows/build-rd-cpp.yml

```diff
@@ -0,0 +1,239 @@
+name: rd-cpp
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  NINJA_VERSION: 1.9.0
+  BUILD_TYPE: Release
+  MAJOR_VERSION: 201.1
+
+jobs:
+  build:
+    env:
+      working-dir: ${{ github.workspace}}${{ matrix.config.SEP }}rd-cpp
+      version: 201.1.${{ github.run_number }}
+      artifact: ${{ matrix.config.artifact }}201.1.${{ github.run_number }}.zip
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+        - {
+            name: "RD CPP Windows Latest MSVC", artifact: "RD-CPP-Windows-MSVC-2019-",
+            os: windows-latest,
+            cc: "cl", cxx: "cl",
+            environment_script: "C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Auxiliary/Build/vcvars64.bat",
+            SEP: \,
+            suffix: win-vs19            
+          }
+        - {
+            name: "RD CPP Windows VS 2017", artifact: "RD-CPP-Windows-MSVC-2017-",
+            os: windows-2016,
+            cc: "cl", cxx: "cl",
+            environment_script: "C:/Program Files (x86)/Microsoft Visual Studio/2017/Enterprise/VC/Auxiliary/Build/vcvars64.bat",
+            SEP: \,
+            suffix: win-vs17
+          }
+#         - {
+#             name: "RD CPP Ubuntu Latest Clang", artifact: "RD-CPP-Linux-Clang-",
+#             os: ubuntu-latest,
+#             cc: "clang", cxx: "clang++",
+#             SEP: /,
+#               suffix: linux-clang
+#           }
+#         - {
+#             name: "RD CPP macOS Latest Clang", artifact: "RD-CPP-macOS-Clang-",
+#             os: macos-latest,
+#             cc: "clang", cxx: "clang++",
+#             SEP: /,
+#               suffix: mac-clang
+#           }
+
+    steps:
+    - uses: actions/checkout@v1
+      
+    - name: Cache submodules
+      uses: actions/cache@v2
+      id: cache-submodules
+      with:
+        path: |
+          rd-cpp/thirdparty/spdlog
+          rd-cpp/thirdparty/googletest
+        key: ${{ matrix.config.os }}-submodules-${{ hashFiles('.gitmodules') }}
+      
+    - name: Get submodules
+      if: steps.cache-submodules.outputs.cache-hit != 'true'
+      run: git submodule update --init --recursive
+
+    - name: Setup .NET Core @ Latest
+      uses: actions/setup-dotnet@v1
+      with:
+        source-url: https://nuget.pkg.github.com/jetbrains/index.json
+      env:
+        NUGET_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+    - name: Cache nuget
+      uses: actions/cache@v1
+      id: cache-nuget
+      with:
+        path: ~/.nuget/packages
+        key: ${{ runner.os }}-nuget-${{ hashFiles('**/packages.lock.json') }}
```

Comment:
Why exactly do we need this? Are we installing any NuGet packages during the build? I thought we only use `nuget pack` and such. If so, then this step may be deleted.

Also, this seems to rely on `pacakages.lock.json` and we definitely aren't using these (yet).

Reply:
I've copy-pasted it from the example for caching nuget as I don't know anything about the way it's being stored. I'm ok with deleting it)